src/components/WalletConnect/CreateTokenRequest.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {
+  StyleSheet,
+  View,
+  ScrollView,
+  TouchableWithoutFeedback,
+  Text,
+  Image,
+} from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigation } from '@react-navigation/native';
+import { numberUtils } from '@hathor/wallet-lib';
+import { t } from 'ttag';
+import {
+  createTokenRetry,
+  createTokenRetryDismiss,
+  setCreateTokenStatusReady,
+  walletConnectAccept,
+  walletConnectReject
+} from '../../actions';
+import { COLORS } from '../../styles/themes';
+import NewHathorButton from '../NewHathorButton';
+import { DappContainer } from './NanoContract/DappContainer';
+import { commonStyles } from './theme';
+import { FeedbackContent } from '../FeedbackContent';
+import { DEFAULT_TOKEN, WALLETCONNECT_CREATE_TOKEN_STATUS } from '../../constants';
+import FeedbackModal from '../FeedbackModal';
+import Spinner from '../Spinner';
+import errorIcon from '../../assets/images/icErrorBig.png';
+import checkIcon from '../../assets/images/icCheckBig.png';
+
+const condRenderData = (
+  attribute,
+  title,
+  separator,
+  formatter = (val) => val,
+) => {
+  if (attribute != null) {
+    return (
+      <>
+        { separator && <View style={commonStyles.cardSeparator} /> }
+        <View>
+          <Text style={styles.property}>{ title }</Text>
+          <Text style={[styles.value, styles.bold]}>
+            { formatter(attribute) }
+          </Text>
+        </View>
+      </>
+    );
+  }
+
+  return null;
+};
+
+function renderBoolean(bool) {
+  return bool ? t`Yes` : t`No`;
+}
+
+export const CreateTokenRequestData = ({ data }) => (
+  <View style={[commonStyles.card, commonStyles.cardSplit]}>
+    <View style={commonStyles.cardSplitContent}>
+      { condRenderData(data.name, t`Token Name`, false) }
+      { condRenderData(data.symbol, t`Token Symbol`, true) }
+      { condRenderData(data.amount, t`Token Amount`, true, numberUtils.prettyValue) }
+      { condRenderData(data.changeAddress, t`Address to send change ${DEFAULT_TOKEN.uid}`, true) }
+      { condRenderData(data.mintAuthorityAddress, t`Address to send the mint authority`, true) }
+      { condRenderData(data.meltAuthorityAddress, t`Address to send the melt authority`, true) }
+      { condRenderData(data.allowExternalMintAuthorityAddress, t`Allow external mint authority addresses?`, true, renderBoolean) }
+      { condRenderData(data.allowExternalMeltAuthorityAddress, t`Allow external melt authority addresses?`, true, renderBoolean) }
+      { condRenderData(data.createMint, t`Create mint authority?`, true, renderBoolean) }
+      { condRenderData(data.createMelt, t`Create melt authority?`, true, renderBoolean) }
+      { condRenderData(data.data, t`Token data`, true, (tokenData) => tokenData.split('\n')) }
```

Comment:
Why do you need this split by `/n`? It will deliver an array, not a string to render correctly the value, right?

Reply:
You are right, fixed, thanks